### ISSUE TITLE
Illegal date after file conversion

### ISSUE THREAD

**Issue by ykong1337** [ixmilia/dxf-rs#37]
I found again that some dxf files still have problems.
The following is a dxf file converted using ```ODAFileConverter```

```
...
$TDCREATE
 40
2305814.964456019
  9
$TDUCREATE
 40
1799402.631122685
  9
...
```

From the source code, after conversion, the creation date of this file is ```2305814```
It is less than the ```julian dublin offset(2415020)``` set by ```dxf-rs```

my code
```rust
fn run() {
    if let Some(file) = rfd::FileDialog::new()
        .add_filter("dxf", &["dxf"])
        .pick_file()
    {
        let _d = Drawing::load_file(file).unwrap();
    }
}
```
Error:
```
thread 'main' panicked at 'attempt to subtract with overflow'
```
Happened here:

chrono::offset::local::windows
```rust
fn system_time_to_tm(sys: &SYSTEMTIME, tm: &mut Tm) {
    tm.tm_sec = sys.wSecond as i32;
    tm.tm_min = sys.wMinute as i32;
    tm.tm_hour = sys.wHour as i32;
    tm.tm_mday = sys.wDay as i32;
    tm.tm_wday = sys.wDayOfWeek as i32;
    tm.tm_mon = (sys.wMonth - 1) as i32;
==> tm.tm_year = (sys.wYear - 1900) as i32;
```

Because this date value needs to be processed and handed over to ```chrono```
So currently it obviously computes a negative number result

My idea is to add a judgment here, if its date is less than 12/31/1899
Just let it assign a value of 1/1/1900

dxf::helper_functions
```rust
fn as_datetime<T>(timezone: &T, date: f64) -> DateTime<T>
    where
        T: TimeZone,
{
    // dates are represented as the fractional number of days elapsed since December 31, 1899.
    let epoch = epoch(timezone);
    let duration = if date == 0.0 {
        ChronoDuration::seconds(0)
    } else {
==>     let duration = f64_to_adjusted_duration(date);
==>     match ChronoDuration::seconds(0).cmp(&duration) {
==>         Ordering::Less => duration,
==>         _ => ChronoDuration::seconds(0)
==>     }
    };
    epoch + duration
}
```

**Comment by brettfo** [comment]
Can you send me the original file that you passed into ODAFileConverter that caused this illegal date to be generated?  If you don't want to post the file here you can email it to me directly at `brett.forsgren@outlook.com` and I'll keep it private.

**Comment by ykong1337** [comment]
> Can you send me the original file that you passed into ODAFileConverter that caused this illegal date to be generated? If you don't want to post the file here you can email it to me directly at `brett.forsgren@outlook.com` and I'll keep it private.

OK, I have sent to your mailbox

**Comment by brettfo** [comment]
Sure enough, the values in the original DWG were also messed up.  I found that AutoCAD represented the first date (`2305814.964456019`) as in the year 1601, and the second date (`1799402.631122685`) was so far out of bounds that it simply chose the current date/time, so it obviously doesn't matter what we do so your suggested change of accepting 1/1/1900 seems perfectly fine.

PR has been merged and the drawing you sent has been deleted.
